src/replication/replication_engine_tests.rs

```diff
@@ -260,29 +260,85 @@ mod tests {
         replicator
     }
 
+    fn fetch_transactions(
+        replicator: &Arc<Replicator>,
+        shard_id: u32,
+        height: u64,
+        message_limit: usize,
+    ) -> Vec<proto::Transaction> {
+        let mut next_page_token = None;
+        let mut transactions = vec![];
+
+        loop {
+            let results = replicator.transactions_for_shard_and_height(
+                shard_id,
+                height,
+                next_page_token.clone(),
+                message_limit,
+            );
+
+            let (results, next_page) = match results {
+                Ok(res) => res,
+                Err(e) => {
+                    panic!(
+                        "Error fetching transactions for shard {} and height {}: {}",
+                        shard_id, height, e
+                    );
+                }
+            };
+
+            transactions.extend(results);
+            next_page_token = next_page;
+
+            if next_page_token.is_none() {
+                break;
+            }
+        }
+
+        transactions
+    }
+
     fn replicate_fids(
         source_engine: &ShardEngine,
         dest_engine: &mut ShardEngine,
         replicator: Arc<Replicator>,
         fids_to_sync: &Vec<u64>,
```

Comment:
Do we need to filter for txns only for these fids somewhere here? I don't see code that does that.